test/trace_processor/diff_tests/stdlib/prelude/args_functions_tests.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env python3
+# Copyright (C) 2025 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from python.generators.diff_tests.testing import Path, DataPath, Metric
+from python.generators.diff_tests.testing import Csv, Json, TextProto
+from python.generators.diff_tests.testing import DiffTestBlueprint
+from python.generators.diff_tests.testing import TestSuite
+
+
+class ArgsFunctions(TestSuite):
+
+  def test_extract_arg(self):
+    return DiffTestBlueprint(
+        trace=TextProto(r"""
+        packet {
+          trusted_packet_sequence_id: 1
+          incremental_state_cleared: true
+          timestamp: 10000000
+          track_event {
+            track_uuid: 1
+            categories: "cat"
+            name: "Event"
+            type: TYPE_INSTANT
+            debug_annotations {
+              name: "key1"
+              string_value: "value1"
+            }
+            debug_annotations {
+              name: "key2"
+              int_value: 42
+            }
+            debug_annotations {
+              name: "key3"
+              double_value: 3.25
+            }
+          }
+        }
+        """),
+        query="""
+        SELECT
+          name,
+          extract_arg(arg_set_id, 'debug.key1') AS string_val,
+          extract_arg(arg_set_id, 'debug.key2') AS int_val,
+          extract_arg(arg_set_id, 'debug.key3') AS double_val,
+          extract_arg(arg_set_id, 'debug.missing_key') AS missing_val
+        FROM slice;
+        """,
+        out=Csv("""
+        "name","string_val","int_val","double_val","missing_val"
+        "Event","value1",42,3.250000,"[NULL]"
+        """))
+
+  def test___intrinsic_arg_set_to_json_simple(self):
```

Comment:
Remove __intrinisic here and below: the fact it's an intrinisic doesn't matter for tests.